app/claim/route.js

```diff
@@ -0,0 +1,31 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  unauthenticatedRedirect: 'signup',
+
+  model: function(params){
+    return {
+      verificationCode: params.verification_code,
+      invitationId: params.invitation_id,
+      type: 'invitation'
+    };
+  },
+
+  actions: {
+    didTransition: function(){
+      var options = this.currentModel;
+      var verification = this.store.createRecord('verification', options);
+
+      verification.save().then( () => {
+        this.replaceWith('index');
+      }, () => {
+        this.controllerFor('claim').set('error', `
+          There was an error accepting this invitation. Perhaps
+          the verification code has expired?
+        `);
```

Comment:
Doing this in `didTransition` allows the template to render — is that the reason for doing this in `didTransition`? I'm not clear on the pros/cons of async in this hook. Can the user get in a weird state by clicking away while this save is still `inFlight`?